From 9c2493d43000a090626b1e8707c3eb3300646695 Mon Sep 17 00:00:00 2001
From: Tingguo Cheng <quic_tingguoc@quicinc.com>
Date: Sun, 29 Sep 2024 12:50:16 +0800
Subject: [PATCH 18/66] arm64: dts: qcom: qcs615: Adds SPMI bus, PMIC and
 peripherals

Add SPMI bus arbiter and include pm8150.dtsi for PMIC peripherals in
pmm6155au which is a variant of pm8150. The enabled peripherals inc-
lude GPIO, RTC, PON, power key and resin-key for QCS615 platform.
The power key and volume down key are controlled by PMIC PON hardware
on pmm6155au.

Signed-off-by: Tingguo Cheng <quic_tingguoc@quicinc.com>
---
 arch/arm64/boot/dts/qcom/qcs615-ride.dts | 35 ++++++++++++++++++++++++
 arch/arm64/boot/dts/qcom/qcs615.dtsi     | 29 ++++++++++++++++++--
 2 files changed, 62 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/qcs615-ride.dts b/arch/arm64/boot/dts/qcom/qcs615-ride.dts
index 4ef969a6af15..5fd46bf991ac 100644
--- a/arch/arm64/boot/dts/qcom/qcs615-ride.dts
+++ b/arch/arm64/boot/dts/qcom/qcs615-ride.dts
@@ -6,6 +6,7 @@
 
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
 #include "qcs615.dtsi"
+#include "pm8150.dtsi"
 / {
 	model = "Qualcomm Technologies, Inc. QCS615 Ride";
 	compatible = "qcom,qcs615-ride", "qcom,qcs615";
@@ -213,6 +214,40 @@ &rpmhcc {
 	clocks = <&xo_board_clk>;
 };
 
+&spmi_bus {
+	pmm6155au_0: pmic@0 {
+
+		pon: pon@800 {
+
+			/delete-property/ mode-bootloader;
+			/delete-property/ mode-recovery;
+
+			pon_pwrkey: pwrkey {
+				status = "okay";
+			};
+
+			pon_resin: resin {
+				linux,code = <KEY_VOLUMEDOWN>;
+				status = "okay";
+			};
+		};
+
+		pmm6155au_0_temp: temp-alarm@2400 {
+			status = "disabled";
+		};
+
+		pmm6155au_0_adc: adc@3100 {
+			status = "disabled";
+		};
+
+		pmm6155au_0_gpios: gpio@c000 {};
+	};
+
+	pmm6155au_1: pmic@1 {
+		status = "disabled";
+	};
+};
+
 &uart0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/qcom/qcs615.dtsi b/arch/arm64/boot/dts/qcom/qcs615.dtsi
index c72759c4f28d..ec02015747e4 100644
--- a/arch/arm64/boot/dts/qcom/qcs615.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs615.dtsi
@@ -553,8 +553,33 @@ sram@c3f0000 {
 			reg = <0x0 0x0c3f0000 0x0 0x400>;
                 };
 
-		apps_smmu: iommu@15000000 {
-			compatible = "qcom,qcs615-smmu-500", "qcom,smmu-500", "arm,mmu-500";
+                spmi_bus: spmi@c440000 {
+                	compatible = "qcom,spmi-pmic-arb";
+                	reg = <0x0 0x0c440000 0x0 0x1100>,
+                	      <0x0 0x0c600000 0x0 0x2000000>,
+                	      <0x0 0x0e600000 0x0 0x100000>,
+                	      <0x0 0x0e700000 0x0 0xa0000>,
+                	      <0x0 0x0c40a000 0x0 0x26000>;
+                	reg-names = "core",
+                	            "chnls",
+                	            "obsrvr",
+                	            "intr",
+                	            "cnfg";
+                	interrupts-extended = <&pdc 1 IRQ_TYPE_LEVEL_HIGH>;
+                	interrupt-names = "periph_irq";
+                	interrupt-controller;
+                	#interrupt-cells = <4>;
+                	#address-cells = <2>;
+                	#size-cells = <0>;
+                	cell-index = <0>;
+                	qcom,channel = <0>;
+                	qcom,ee = <0>;
+               	};
+
+               	apps_smmu: iommu@15000000 {
+               		compatible = "qcom,qcs615-smmu-500",
+               		             "qcom,smmu-500",
+               		             "arm,mmu-500";
 			reg = <0x0 0x15000000 0x0 0x80000>;
 			#iommu-cells = <2>;
 			#global-interrupts = <1>;
-- 
2.34.1

